### PR TITLE
Extend adapter interface with Name function

### PR DESCRIPTION
**Pull request by lafriks**
Refs #345 

~~I went with `DBType` name as this is also what for example xorm uses and it's more descriptive~~

### CODE REVIEW

**Comment by lafriks** [comment]
Should be ready to merge, once this is merged I will update deps to go-rel/sql and later on on all adapters

**Review by Fs02 — APPROVED** [review]
LGTM, Thanks

**Pull request merged**
